src/analysis/special_functions.rs

```diff
@@ -28,40 +34,91 @@ impl FromStr for Type {
             "free" | "destroy" => Ok(Free),
             "is_equal" => Ok(Equal),
             "ref" | "ref_" => Ok(Ref),
-            "to_string" => Ok(ToString),
+            "to_string" => Ok(Display),
             "unref" => Ok(Unref),
             "hash" => Ok(Hash),
             _ => Err(format!("Unknown type '{}'", s)),
         }
     }
 }
 
-pub type Infos = BTreeMap<Type, String>; // Type => glib_name
+impl Type {
+    pub(crate) fn extract(s: &str) -> Option<Self> {
+        s.parse().ok().or_else(|| match s {
+            "get_name" | "name" => Some(Self::Display),
+            _ => None,
+        })
+    }
+}
 
-fn update_func(func: &mut FuncInfo, type_: Type) -> bool {
-    if func.visibility != Visibility::Comment {
-        func.visibility = visibility(type_, func.parameters.c_parameters.len());
+#[derive(Debug, Clone)]
+pub struct Info {
+    pub glib_name: String,
+    pub returns_static_ref: bool,
+    pub version: Option<Version>,
+}
+
+pub type Infos = BTreeMap<Type, Info>;
+
+/// Returns true on functions that take an instance as single argument and
+/// return a string as result.
+fn is_stringify(func: &mut FuncInfo, parent_type: &LibType, obj: &GObject) -> bool {
+    if func.parameters.c_parameters.len() != 1 {
+        return false;
     }
-    // I assume `to_string` functions never return `NULL`
-    if type_ == Type::ToString {
-        if let Some(par) = func.ret.parameter.as_mut() {
-            *par.nullable = false;
-        }
-        if func.visibility != Visibility::Private {
+    if !func.parameters.c_parameters[0].instance_parameter {
+        return false;
+    }
+
+    if let Some(ret) = func.ret.parameter.as_mut() {
+        if ret.typ != TypeId::tid_utf8() {
             return false;
         }
+
+        if func.name == "to_string" {
+            // Rename to to_str to make sure it doesn't clash with ToString::to_string
+            func.name = "to_str".to_owned();
+
+            // As to not change old code behaviour, assume non-nullability outside
+            // enums and flags only, and exclusively for to_string. Function inside
+            // enums and flags have been appropriately marked in Gir.
+            if !obj.trust_return_value_nullability
+                && !matches!(parent_type, LibType::Enumeration(_) | LibType::Bitfield(_))
+            {
+                *ret.nullable = false;
+            }
+        }
+
+        !*ret.nullable
+    } else {
+        false
+    }
+}
+
+fn update_func(func: &mut FuncInfo, type_: Type) -> bool {
```

Comment:
`type_` is not used anymore :)

Reply:
Huh, it's passed into `visibility(type_) below:

```rust
fn update_func(func: &mut FuncInfo, type_: Type) -> bool {
    if func.visibility != Visibility::Comment {
        func.visibility = visibility(type_);
    }
    true
}
```

Reply:
Ah sorry, was confused by the diff :)